pkg/auth/api/user/user_actions.go

```diff
@@ -134,6 +138,20 @@ func (h *Handler) setPassword(actionName string, action *types.Action, request *
 		return errors.New("Invalid password")
 	}
 
+	userID := request.Request.Header.Get("Impersonate-User")
+	if userID == "" {
+		return errors.New("can't find user")
+	}
+
+	user, err := h.UserClient.Get(userID, v1.GetOptions{})
```

Comment:
I'm surprised we don't use a userLister in this file. Maybe we can't risk the data not being up to date.